site/_posts/2023-03-10-growing-is-hard.md

```diff
@@ -0,0 +1,63 @@
+
+The downside is that you rarely develop the same level of experiences and skill that you would by sticking around, learning to work through conflict, mastering a code base and learning how to stay productive in spite of company politics.
+
+Unfortunately this doesn't seem to be the sentiment shared by many engineers. 
+
+### The problem
+By the time an employee is beginning to produce and become competent - they have one foot out the door looking for their next adventure. 
+
+There are various ways to overcome this problem.
+
+Venture capital backed companies solve this by hiring many engineers at once. Using this strategy, even if a few leave, there is nearly always overlap in tenure so that the combined skills on the team never go to zero.
+
+At FusionAuth, we've taken a different path. things a little different. We've never taken capital, we've built a profitable business w/out debt. We hire when we can afford to do so.
+
+Hiring slow, means we may only be able to afford to hire a new engineer every 12-18 months. If the average tenure is less than your hiring cadence - you'll never get ahead of the attrition curve.
+
+It is sort of like filling a tub without plugging the drain. Unless you have great water pressure , the tub will drain faster than you can fill it. It is just difficult to get ahead, and you end up sitting cold and naked in the tub. The analogy does fall apart a bit at the end.   
+
+The result is that it leaves the technical founders holding the bag every time. This cycle repeats and can become demoralizing when the time and effort that is spent imparting the skills necessary to continue and grow the business are never preserved.
+
+### Light at the end of the tunnel
+
+Thankfully, we have turned the corner on this issue and begun to dig our way out of the whole. As FusionAuth can continued to grow, we can now afford to pay competitive salaries, and we also have enough revenue to hire in bulk. Hiring more than one engineer at a time is helpful in several ways. 
+
+The first is that the new team feels a sense of community as they all start from zero together. They do not have to compare their skill set or knowledge of the product with someone that has been doing it for 10 years, and instead can compare to another that is learning the same things.
+
+The second - which is most exciting to me - is that we are now filling the tub four times as fast!
+
+### Lessons learned
+
+Seeing mistakes in hindsight is not difficult. I would do many things differently given the opportunity. 
+
+One of our biggest mistakes is thinking we could not afford to hire senior engineers. It can be difficult to hire these engineers early in a company because their salary requirements will likely be more than you pay yourself. But the reality is that they can produce 5-10x of an entry level engineer, and their salary not even close to 5-10x, and more likely 2x. 
```

Comment:
I'm not a fan of the productivity comparison. Partly because I think the "10x engineer" thing is dumb.

Reply:
This isn't so much a "10x-er" statement - but the reality of what an experienced engineer can do in contrast to a brand new engineer. Open to better ways to say it -but there certainly is a difference in productivity.

Reply:
I'd anchor this in personal experience, and maybe leave off the numbers (which I assume are anecdotal). Maybe something like:

```
But the reality is that a senior engineer can get up to speed more quickly and therefore can produce far more value (through code and processes) than someone with less experience. The value they bring is far more than the extra cost. For example, we had a senior engineer take on a task similar to <X>, whereas before a less senior engineer had a hard time doing a similar task.
```

Reply:
Not sure about if you are going to call out anyone, so maybe ignore the 'for example' above if you can't do so without saying unkind things about someone.